src/game_updater/GameUpdater.cpp

```diff
@@ -185,6 +186,14 @@ void GameUpdater::UpdatePlayer(std::shared_ptr<GameState> game_state) {
   } else {
     // player is in the air, apply gravity
     player->SetYVelocity(previous_player_velocity - PLAYER_GRAVITY);
+    // if player is close to falling, allow double jump
+    if (player->GetYVelocity() < PLAYER_JUMP_VELOCITY / 3.0 &&
+          player->GetDoubleJump() &&
+          ImGui::GetIO().KeysDown[GLFW_KEY_SPACE]) {
```

Comment:
Sound like a good idea. I would also like the player to double jump when the key is pressed rather than when it is held down so that the player doesn't go straight into a double jump if the spacebar isn't released after a jump. If that's not possible without key events based on interrupts this is fine for now.